### PR TITLE
chore: add a histogram for tx coefficients

### PR DESCRIPTION
**Pull request by darrenvechain**
# Description

Add a histogram metric for tx coefficients

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: consensus/validator.go [hunk 1]
```diff
@@ -328,6 +328,7 @@ func (c *Consensus) verifyBlock(blk *block.Block, state *state.State, blockConfl
 		totalGasUsed += receipt.GasUsed
 		receipts = append(receipts, receipt)
 		processedTxs[tx.ID()] = receipt.Reverted
+		metricCoefBucket().Observe(int64(tx.GasPriceCoef()))
```
Comment (otherview) [inline consensus/validator.go:331]: What happens if this block is invalid ?
What if this metric was set somewhere near the `metricBlockProcessedxxx` metrics ?

Comment (darrenvechain) [reply consensus/validator.go]: Fixed

**Comment by libotony** [comment]
IMO, this one is better collected off chain, like https://github.com/vechain/vestatus

**Comment by libotony** [comment]
> > IMO, this one is better collected off chain, like https://github.com/vechain/vestatus
> 
> Shouldn't thor allow all data that can be collected from the node to be available ? I think this falls similar to the block gas and tx gas metrics.

I think the scope of thor metrics should be limited to the runtime(real time) info. Block/Tx related is better collect off the client, the `time` of the 'facts' related metrics was born with it(block time) rather than the time it was collected.

**Comment by darrenvechain** [comment]
Closing as we can do this with historical data

**Pull request closed without merging**